MonoGame.Framework/Graphics/Model.cs

```diff
@@ -1,44 +1,80 @@
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
 using System;
-using System.Diagnostics;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 
 namespace Microsoft.Xna.Framework.Graphics
 {
-	public class Model
+    /// <summary>
+    /// Represents a 3D-Model.
+    /// </summary>
+	public sealed class Model
 	{
 		private static Matrix[] sharedDrawBoneMatrices;
 		
 		private GraphicsDevice graphicsDevice;
-        private GraphicsDevice GraphicsDevice { get { return this.graphicsDevice; } }
-
-		// Summary:
-		//     Gets a collection of ModelBone objects which describe how each mesh in the
-		//     Meshes collection for this model relates to its parent mesh.
-		public ModelBoneCollection Bones { get; private set; }
-		//
-		// Summary:
-		//     Gets a collection of ModelMesh objects which compose the model. Each ModelMesh
-		//     in a model may be moved independently and may be composed of multiple materials
-		//     identified as ModelMeshPart objects.
-		public ModelMeshCollection Meshes { get; private set; }
-		//
-		// Summary:
-		//     Gets the root bone for this model.
-		public ModelBone Root { get; set; }
-		//
-		// Summary:
-		//     Gets or sets an object identifying this model.
-		public object Tag { get; set; }
-
-		public Model()
+
+        /// <summary>
+        /// A collection of <see cref="ModelBone"/> objects which describe how each mesh in the
+        /// mesh collection for this model relates to its parent mesh.
+        /// </summary>
+        public ModelBoneCollection Bones { get; private set; }
+
+        /// <summary>
+        /// A collection of <see cref="ModelMesh"/> objects which compose the model. Each <see cref="ModelMesh"/>
+        /// in a model may be moved independently and may be composed of multiple materials
+        /// identified as <see cref="ModelMeshPart"/> objects.
+        /// </summary>
+        public ModelMeshCollection Meshes { get; private set; }
+
+        /// <summary>
+        /// Root bone for this model.
+        /// </summary>
+        public ModelBone Root { get; set; }
+
+        /// <summary>
+        /// Custom attached object.
+        /// <remarks>
+        /// Skinning data is example of attached object for model.
+        /// </remarks>
+        /// </summary>
+        public object Tag { get; set; }
+
+        /// <summary>
+        /// Sets custom effect for all meshes in this model.
+        /// </summary>
+        /// <param name="effect">Specified <see cref="Effect"/>.</param>
+        /// <remarks>
+        /// This call is equivalent to
+        /// <code>
+        /// foreach (var mesh in model.Meshes)
+        /// {
+        ///    foreach (var part in mesh.MeshParts)
+        ///    {
+        ///       part.Effect = effect;
+        ///    }
+        /// }
+        /// </code>
+        /// </remarks>
+        public void SetCustomEffect(Effect effect)

```

Comment:
This function seems completely unnecessary to me.  Especially we should not copy the code from a function into the function's own documentation.

@KonajuGames ?